control_plane/src/compute.rs

```diff
@@ -306,34 +268,30 @@ impl PostgresNode {
     // Connect to a page server, get base backup, and untar it to initialize a
     // new data directory
     fn setup_pg_conf(&self, auth_type: AuthType) -> Result<()> {
-        File::create(self.pgdata().join("postgresql.conf").to_str().unwrap())?;
+        File::create(self.pgdata().join("postgresql.conf"))?;
 
         // wal_log_hints is mandatory when running against pageserver (see gh issue#192)
         // TODO: is it possible to check wal_log_hints at pageserver side via XLOG_PARAMETER_CHANGE?
-        self.append_conf(
-            "postgresql.conf",
-            &format!(
-                "max_wal_senders = 10\n\
-                 wal_log_hints = on\n\
-                 max_replication_slots = 10\n\
-                 hot_standby = on\n\
-                 shared_buffers = 1MB\n\
-                 fsync = off\n\
-                 max_connections = 100\n\
-                 wal_sender_timeout = 0\n\
-                 wal_level = replica\n\
-                 listen_addresses = '{address}'\n\
-                 port = {port}\n",
-                address = self.address.ip(),
-                port = self.address.port()
-            ),
-        )?;
+
+        use std::fmt::Write;
+        let mut conf = String::new();
+        writeln!(conf, "max_wal_senders = 10")?;
+        writeln!(conf, "wal_log_hints = on")?;
+        writeln!(conf, "max_replication_slots = 10")?;
+        writeln!(conf, "hot_standby = on")?;
+        writeln!(conf, "shared_buffers = 1MB")?;
+        writeln!(conf, "fsync = off")?;
+        writeln!(conf, "max_connections = 100")?;
+        writeln!(conf, "wal_sender_timeout = 0")?;
+        writeln!(conf, "wal_level = replica")?;
+        writeln!(conf, "listen_addresses = '{}'", self.address.ip())?;
+        writeln!(conf, "port = {}\n", self.address.port())?;
```

Comment:
I wonder, what do we win here? IMO, those numerous `writeln!(conf,` don't look very appealing. If you're annoyed by string escapes, we could instead concat string literals like this:

```rust
&format!(
  concat![
    "here goes:\n",
    "foo is {foo}\n",
    "bar is {bar}\n",
  ],
  foo = 1,
  bar = 2,
)
```

We could go further and define a small macro to append those newlines for us:

```rust
macro_rules! lines {
    ($($literal:expr),+ $(,)?) => {
        concat![
            $(
                $literal,
                "\n",
            )+
        ]
    }
}
```

```rust
    println!("'{}'", format!(
        lines!(
            "those are the lines:",
            "foo is {foo}",
            "bar is {bar}"
        ),
        foo = 1,
        bar = "value",
    ));
```

Reply:
I'd like to have each parameter on the same line as the option. So instead of
```
"foo={foo}"
"bar={bar}"
,
  foo=123
  bar=456
```
I want to have something like this:
```
"foo={}", 123
"bar={}", 456
```

IMHO the latter is more readable, and produces simpler diffs when you add or remove code for a setting, as you only need to add/remove one source line.

Reply:
If line has exactly one value that's formatted into it, you can use this:

```rs
macro_rules! conf_lines {
    ($($format:literal, $value:expr,)*) => {{
        format!(concat!( $($format, "\n",)* ), $($value,)*)
    }};
}

println!("{}", conf_lines!(
    "foo={}", 123,
    "bar={}", 456,
))
```

Unfortunately we'd have to use proc macros to allow variable numbers of format arguments, which is probably overkill.